exasol_integration_test_docker_environment/test/test_test_env_reuse.py

```diff
@@ -0,0 +1,103 @@
+            docker_username=None,
+            tag_prefix="",
+            kind="target"
+        )
+
+    def run_spawn_test_env(self):
+        result = None
+        self.setup_luigi_config()
+        luigi_utils.set_job_id(SpawnTestEnvironment)
+        task = SpawnTestEnvironment(reuse_database_setup=True,
+                                    reuse_database=True,
+                                    reuse_test_container=True,
+                                    external_exasol_xmlrpc_host="",
+                                    external_exasol_db_host="",
+                                    external_exasol_xmlrpc_port=0,
+                                    external_exasol_db_user="",
+                                    external_exasol_db_password="",
+                                    external_exasol_xmlrpc_user="",
+                                    external_exasol_xmlrpc_password="",
+                                    external_exasol_xmlrpc_cluster_name="",
+                                    external_exasol_bucketfs_write_password="",
+                                    environment_type=EnvironmentType.docker_db,
+                                    environment_name=self.env_name(),
+                                    docker_db_image_version=self.docker_db_version_parameter,
+                                    docker_db_image_name="exasol/docker-db"
+                                    )
+        try:
+            success = luigi.build([task], workers=1, local_scheduler=True, log_level="INFO")
+            if success:
+                result = task
+            else:
+                task.cleanup(False)
+                Exception("Task failed")
+        except Exception as e:
+            task.cleanup(False)
+        return result
+
+    def test_initial_reuse_database_setup_populates_data(self):
+        username = SpawnTestEnvironment.DEFAULT_DB_USER
+        password = SpawnTestEnvironment.DEFAULT_DATABASE_PASSWORD
+        task = process_spawn_utils.run_in_process(self.run_spawn_test_env)
+        test_environment_info = task.get_return_object()
+        test_container = self._client.containers.get(test_environment_info.test_container_info.container_name)
+        database_host = test_environment_info.database_info.host
+        database_port = test_environment_info.database_info.db_port
+        q = "SELECT TABLE_NAME FROM SYS.EXA_ALL_TABLES WHERE TABLE_SCHEMA='TEST';"
+        cmd = f"""$EXAPLUS -c '{database_host}:{database_port}' -u '{username}' -p '{password}' -sql \\\"{q}\\\""""
+        bash_cmd = f"""bash -c "{cmd}" """
+        exit_code, output = test_container.exec_run(cmd=bash_cmd)
+        output_str = output.decode('utf-8')
+
+        # TODO read /tests/test/import.sql and apply a regular expression to
+        # get all table names and compare then one-by-one
+        self.assertIn("ENGINETABLE", [output_entry.strip() for output_entry in output_str.split(sep='\n')])
+        task.cleanup(True)
+
```

Comment:
we also need to test the case, were you reuse an actually started db

Reply:
Done

Reply:
Done